rust/lance-io/src/encodings/binary.rs

```diff
@@ -224,7 +224,7 @@ impl<'a, T: ByteArrayType> BinaryDecoder<'a, T> {
                 .null_bit_buffer(null_buf);
         }
 
-        let buf = bytes.into();
+        let buf = Buffer::from_vec(bytes.to_vec());
```

Comment:
This triggers a data copy (and could be the source of your alignment issues).  What changed in arrow-rs to necessitate this change?  It should still be possible to go from `Bytes` to `Buffer` with zero-copy somehow.

Reply:
Changed to using `bytes.to_byte_slice().into()` as suggested in that comment but alas still seeing alignment issues:
```
---- index::vector::ivf::tests::test_create_ivf_hnsw_with_empty_partition stdout ----
thread 'index::vector::ivf::tests::test_create_ivf_hnsw_with_empty_partition' panicked at /home/jleibs/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-buffer-53.3.0/src/buffer/scalar.rs:133:42:
Memory pointer is not aligned with the specified scalar type
```

Reply:
That's still going to trigger a copy it seems.  I'll take a look real quick.

Reply:
> That's still going to trigger a copy it seems.

Is it?

My reads of both:
-  `impl From<Bytes> for Vec<u8>`
- `Buffer::from_vec()`

Look like they should both be zero-copy.

Reply:
Sorry, I saw `bytes.to_byte_slice().into()` which I think is not zero-copy.  You are right that `Buffer::from_vec(bytes.into())` should be zero-copy.

Reply:
Actually, no `Buffer::from_vec(bytes.into())` will not be zero copy because `bytes.into()` is not zero-copy

Reply:
I'm guessing your fix was probably ok here.  The root cause was probably just the `deep_copy_buffer` change.

Reply:
Aha. Those new routines are much more clear. Thanks.